WowPacketParserModule.V2_5_1_38707/Parsers/QueryHandler.cs

```diff
@@ -0,0 +1,290 @@
+            creature.WidgetSetID = packet.ReadInt32("WidgetSetID");
+            creature.WidgetSetUnitConditionID = packet.ReadInt32("WidgetSetUnitConditionID");
+
+            if (titleLen > 1)
+                creature.SubName = packet.ReadCString("Title");
+
+            if (titleAltLen > 1)
+                creature.TitleAlt = packet.ReadCString("TitleAlt");
+
+            if (cursorNameLen > 1)
+                creature.IconName = packet.ReadCString("CursorName");
+
+            for (uint i = 0; i < questItems; ++i)
+            {
+                CreatureTemplateQuestItem questItem = new CreatureTemplateQuestItem
+                {
+                    CreatureEntry = (uint)entry.Key,
+                    Idx = i,
+                    ItemId = (uint)packet.ReadInt32<ItemId>("QuestItem", i)
+                };
+
+                Storage.CreatureTemplateQuestItems.Add(questItem, packet.TimeSpan);
+            }
+
+            packet.AddSniffData(StoreNameType.Unit, entry.Key, "QUERY_RESPONSE");
+
+            Storage.CreatureTemplates.Add(creature.Entry.Value, creature, packet.TimeSpan);
+
+            if (ClientLocale.PacketLocale != LocaleConstant.enUS)
+            {
+                CreatureTemplateLocale localesCreature = new CreatureTemplateLocale
+                {
+                    ID = (uint)entry.Key,
+                    Name = creature.Name,
+                    NameAlt = creature.FemaleName,
+                    Title = creature.SubName,
+                    TitleAlt = creature.TitleAlt
+                };
+
+                Storage.LocalesCreatures.Add(localesCreature, packet.TimeSpan);
+            }
+
+            ObjectName objectName = new ObjectName
+            {
+                ObjectType = StoreNameType.Unit,
+                ID = entry.Key,
+                Name = creature.Name
+            };
+            Storage.ObjectNames.Add(objectName, packet.TimeSpan);
+        }
+
+        [Parser(Opcode.SMSG_QUERY_PET_NAME_RESPONSE)]
+        public static void HandlePetNameQueryResponse(Packet packet)
+        {
+            packet.ReadPackedGuid128("PetID");
+
+            var hasData = packet.ReadBit("HasData");
+            if (!hasData)
+                return;
+
+            var len = packet.ReadBits(8);
+            packet.ReadBit("HasDeclined");
+
+            const int maxDeclinedNameCases = 5;
+            var declinedNameLen = new int[maxDeclinedNameCases];
+            for (var i = 0; i < maxDeclinedNameCases; ++i)
+                declinedNameLen[i] = (int)packet.ReadBits(7);
+
+            for (var i = 0; i < maxDeclinedNameCases; ++i)
+                packet.ReadWoWString("DeclinedNames", declinedNameLen[i], i);
+
+            packet.ReadTime64("Timestamp");
+            packet.ReadWoWString("Petname", len);
+        }
+
+        [HasSniffData]
+        [Parser(Opcode.SMSG_QUERY_GAME_OBJECT_RESPONSE)]
+        public static void HandleGameObjectQueryResponse(Packet packet)
```

Comment:
Same as `V8_0_1_27101.Parsers.HandleGameObjectQueryResponse` (might need logic adjustment in determining ContentTuningId)

Reply:
You are right.